x-pack/plugins/ml/public/maps/anomaly_source.tsx

```diff
@@ -0,0 +1,356 @@
+    registerCancelCallback: (callback: () => void) => void,
+    isRequestStillActive: () => boolean
+  ): Promise<GeoJsonWithMeta> {
+    const results = await getResultsForJobId(
+      AnomalySource.mlResultsService,
+      this._descriptor.jobId,
+      this._descriptor.typicalActual,
+      searchFilters
+    );
+
+    return {
+      data: results,
+      meta: {
+        // Set this to true if data is incomplete (e.g. capping number of results to first 10k)
+        areResultsTrimmed: false,
+      },
+    };
+  }
+
+  canFormatFeatureProperties(): boolean {
+    return false;
+  }
+
+  cloneDescriptor(): AnomalySourceDescriptor {
+    return {
+      type: this._descriptor.type,
+      jobId: this._descriptor.jobId,
+      typicalActual: this._descriptor.typicalActual,
+    };
+  }
+
+  createField({ fieldName }: { fieldName: string }): IField {
+    if (fieldName !== 'record_score') {
+      throw new Error('Record score field name is required');
+    }
+    return new RecordScoreField({ source: this });
+  }
+
+  async createFieldFormatter(field: IField): Promise<FieldFormatter | null> {
+    return null;
+  }
+
+  destroy(): void {}
+
+  getApplyGlobalQuery(): boolean {
+    return false;
+  }
+
+  getApplyForceRefresh(): boolean {
+    return false;
+  }
+
+  getApplyGlobalTime(): boolean {
+    return true;
+  }
+
+  async getAttributions(): Promise<Attribution[]> {
+    return [];
+  }
+
+  async getBoundsForFilters(
+    boundsFilters: object,
+    registerCancelCallback: (callback: () => void) => void
+  ): Promise<MapExtent | null> {
+    return null;
+  }
+
+  async getDisplayName(): Promise<string> {
+    return i18n.translate('xpack.ml.maps.anomalySource.displayLabel', {
+      defaultMessage: '{typicalActual} for {jobId}',
+      values: {
+        typicalActual: this._descriptor.typicalActual,
+        jobId: this._descriptor.jobId,
+      },
+    });
+  }
+
+  getFieldByName(fieldName: string): IField | null {
+    if (fieldName === 'record_score') {
+      return new RecordScoreField({ source: this });
+    }
+    return null;
+  }
+
+  getSourceStatus() {
+    return { tooltipContent: null, areResultsTrimmed: false };
+  }
+
+  getType(): string {
+    return this._descriptor.type;
+  }
+
+  isMvt() {
+    return true;
+  }
+
+  showJoinEditor(): boolean {
+    // Ignore, only show if joins are enabled for current configuration
+    return false;
+  }
+
+  getFieldNames(): string[] {
+    return ['record_score'];
+  }
+
+  async getFields(): Promise<IField[]> {
+    return [new RecordScoreField({ source: this })];
+  }
+
+  getGeoGridPrecision(zoom: number): number {
+    return 0;
+  }
+
+  isBoundsAware(): boolean {
+    return false;
+  }
+
+  async getImmutableProperties(): Promise<ImmutableSourceProperty[]> {
+    return [
+      {
+        label: i18n.translate('xpack.ml.maps.anomalySourcePropLabel', {
+          defaultMessage: 'Job Id',
+        }),
+        value: this._descriptor.jobId,
+      },
```

Comment:
Consider adding second object with hyperlink to the anomaly-explorer page in ML based on the job-Id.

e.g. this is some sample code.

https://github.com/elastic/kibana/blob/6a3d194b96deebe145c873895f59021eca3a3725/x-pack/plugins/maps/public/classes/sources/ems_file_source/ems_file_source.tsx#L156-L162

So for ML, it'd be something like:

```
 return [
      {
        label: i18n.translate('xpack.ml.maps.anomalySourcePropLabel', {
          defaultMessage: 'Job Id',
        }),
        value: this._descriptor.jobId,
        link: `http://localhost:5601/hen/app/ml/explorer?_g={jobIds:[%22${this._jobId}%22]` // todo: this link should be constructed with `core.http.basePath.prepend()` and utilities there are on ML-side to derive this URL
      },
```

Reply:
Added for a follow up 👍